cmd/util/cmd/execution-state-extract/execution_state_extract.go

```diff
@@ -86,7 +86,6 @@ func extractExecutionState(
 			extractionReportName: reporters.NewExportReporter(log,
 				chain,
 				func() flow.StateCommitment { return targetHash },
-				func() flow.StateCommitment { return flow.StateCommitment(newState) },
```

Comment:
This is the bug. 

This callback is supposed to read the newState after the migration unless the callback is called AFTER the migration and the newState is overwritten by L104.

The callback was called after the migration, but before the newState is returned to be used to overwrite thew `newState`. Therefore,the callback is still reading the old value.

To fix it, we simply pass the statecommitment to each reporter, so each reporter can decide whether to use it or not.